trench/models.py

```diff
@@ -74,6 +77,19 @@ class MFAMethod(Model):
     class Meta:
         verbose_name = _("MFA Method")
         verbose_name_plural = _("MFA Methods")
+        constraints = (
+            UniqueConstraint(
+                condition=Q(is_primary=True),
+                fields=("user",),
+                name="unique_user_is_primary",
+            ),
+            CheckConstraint(
+                check=(Q(is_primary=True) & Q(is_active=True))
+                | (Q(is_primary=False) & Q(is_active=True))
+                | (Q(is_primary=False) & Q(is_active=False)),
+                name="primary_is_active",
```

Comment:
Good idea with adding a check constraint on a database level! 👍 

There's one thing though - you probably can simplify the query here:
```python
# query 1
(is_primary=True & is_active=True) | (is_primary=False & is_active=True) | (is_primary=False & is_active=False)

# query 2
(is_primary=True & is_active=True) | is_primary=False
```

They both give the same result.

Here's a simulation of both queries with all possible combinations of values for `is_primary` and `is_active`:
![image](https://user-images.githubusercontent.com/53468809/202408765-d29b4651-29f4-43eb-8f06-33d3e5d09582.png)

Reply:
Fixed